tests/test_discovery.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+import subprocess  # noqa: S404
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import virtualenv
+
+from pipdeptree.__main__ import main
+
+if TYPE_CHECKING:
+    import pytest
+
+
+def test_local_only(
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+    capfd: pytest.CaptureFixture[str],
+) -> None:
+    prefix = str(tmp_path / "venv")
+    result = virtualenv.cli_run([str(tmp_path / "venv"), "--activators", ""])
+    pip_path = str(result.creator.exe.parent / "pip")
+    subprocess.run(
+        [pip_path, "install", "wrapt", "--prefix", prefix],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+        check=False,
+    )
+    cmd = [str(result.creator.exe.parent / "python3")]
+    monkeypatch.chdir(tmp_path)
+    cmd += ["--local-only"]
+    monkeypatch.setattr(sys, "prefix", [str(tmp_path / "venv")])
+    monkeypatch.setattr(sys, "argv", cmd)
```

Comment:
Hmm, this is a good test but I think we can avoid network calls by adding some a fake distribution in the virtual environment's site-packages directory. I did it myself and `importlib` doesn't seem to mind me doing the following:
- Create a .dist-info directory in the venv's site-packages directory (e.g. I made one called help-1.2.5.dist-info")
- Add a fake METADATA file. You need only 3 [fields](https://packaging.python.org/en/latest/specifications/core-metadata/) to include in this file.

My pipdeptree output (don't mind `packaging` I was messing around with its version):
```console
(venv) $ pipdeptree
help==1.2.5
pipdeptree==2.16.3.dev27+g757c8ab.d20240329
├── packaging [required: ==20.0, installed: 20.0]
│   ├── pyparsing [required: >=2.0.2, installed: 3.1.2]
│   └── six [required: Any, installed: 1.16.0]
└── pip [required: >=23.1.2, installed: 24.0]
setuptools==69.1.1
wheel==0.42.0

Also, do we need to do `monkeypatch.chdir(tmp_path)`?

Reply:
Wow, that’s really a good idea. You are right, we should better not use network calls in tests. I will try this latter tonight. I am away from home today, so the replies may be a little delayed, sorry.

Reply:
Done.